From bfc38055c1c295da28ddf925cf4ec9eb48ef5e84 Mon Sep 17 00:00:00 2001
From: Guennadi Liakhovetski <g.liakhovetski@gmx.de>
Date: Mon, 24 Jun 2013 05:13:51 -0300
Subject: [PATCH 0689/1302] [media] V4L2: fix compilation if CONFIG_I2C is
 undefined

i2c_verify_client() is only available, if I2C is enabled. Fix v4l2-async.c
compilation if I2C is disabled.

Signed-off-by: Guennadi Liakhovetski <g.liakhovetski@gmx.de>
Signed-off-by: Mauro Carvalho Chehab <mchehab@redhat.com>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 drivers/media/v4l2-core/v4l2-async.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/v4l2-core/v4l2-async.c b/drivers/media/v4l2-core/v4l2-async.c
index c80ffb4..aae2417 100644
--- a/drivers/media/v4l2-core/v4l2-async.c
+++ b/drivers/media/v4l2-core/v4l2-async.c
@@ -24,11 +24,15 @@
 
 static bool match_i2c(struct device *dev, struct v4l2_async_subdev *asd)
 {
+#if IS_ENABLED(CONFIG_I2C)
 	struct i2c_client *client = i2c_verify_client(dev);
 	return client &&
 		asd->bus_type == V4L2_ASYNC_BUS_I2C &&
 		asd->match.i2c.adapter_id == client->adapter->nr &&
 		asd->match.i2c.address == client->addr;
+#else
+	return false;
+#endif
 }
 
 static bool match_platform(struct device *dev, struct v4l2_async_subdev *asd)
-- 
1.8.3.2

